File: onebot-client/src/main/java/cn/evolvefield/onebot/client/listener/EnableEventListener.java
package cn.evolvefield.onebot.client.listener;

/**
 * 提供是否开启插件
 * @param <T>
 */
public abstract class EnableEventListener<T> implements EventListener<T> {

    private Boolean enable = true;//默认开启

    public void setEnable(Boolean enable) {
        this.enable = enable;
    }

    public Boolean getEnable() {
        return enable;
    }

    public Boolean enable() {
        return enable;
    }
}

File: onebot/src/main/kotlin/cn/evolvefield/onebot/client/listener/EnableEventListener.kt
package cn.evolvefield.onebot.client.listener

import cn.evole.onebot.sdk.event.Event

/**
 * 提供是否开启插件
 * @param <T>
</T> */
abstract class EnableEventListener<T : Event> : EventListener<T> {
    var enable = true //默认开启
    fun enable(): Boolean {
        return enable
    }
}